### PR TITLE
[ENG-50123]: Fix the vulnerabilities related to request package for mimecast

### PR DESCRIPTION
**Pull request by rakhimundhada15**
1. Fix the vulnerabilities related to request package for mimecast
2. Bump the salesforce package missed in last  [pr](https://github.com/alertlogic/paws-collector/pull/344/files#diff-8e2fe486da232865ad490e3e034fa0da6a9b0e62c870b3bb4a56571610ec88c3L3)

### CODE REVIEW

File: collectors/mimecast/utils.js [hunk 1]
```diff
@@ -30,41 +30,27 @@ function getAPILogs(authDetails, state, accumulator, maxPagesPerInvocation) {
 
                 AlLogger.debug(`MIME00009 calling url: ${url}`);
                 const tempPayload = {
+                    method: 'POST',
                     url: url,
                     headers: requestHeaders,
-                    body: JSON.stringify(applicationDetails.payload)
+                    data: applicationDetails.payload,
```
Comment (imranalisyed506) [inline collectors/mimecast/utils.js:36]: can you remove extra comma

File: collectors/mimecast/utils.js [hunk 2]
```diff
@@ -30,41 +30,27 @@ function getAPILogs(authDetails, state, accumulator, maxPagesPerInvocation) {
...
-                        return reject(response);
-                    }
-
+                const payloadData = applicationDetails.encoding === null || applicationDetails.encoding ? { ...tempPayload, responseType: 'arraybuffer', encoding: applicationDetails.encoding } : tempPayload;
+                axios.request(payloadData).then(response => {
                     if (!applicationDetails.compress) {
-                        try {
-                            body = JSON.parse(body);
-                        } catch (exception) {
-                            AlLogger.error("MIME000010 Error parsing response. ", body);
```
Comment (imranalisyed506) [inline collectors/mimecast/utils.js:52]: hope this is working fine with normal logs and logs in zip file? can you check when collector is deployed or in local

**Review by imranalisyed506 — COMMENTED** [review]
minor comments otherwise LGTM

File: collectors/mimecast/utils.js (hunk 2 above)
Comment (rakhimundhada15) [reply collectors/mimecast/utils.js:52]: yes. I have checked locally. for  SiemLogs we get the zip file then we have method unzipBufferInMemory  and for other  streams we don’t need the parse after axios api changes

**Review by imranalisyed506 — APPROVED** [review]
LGTM

**Pull request merged**